zulipterminal/ui_tools/views.py

```diff
@@ -305,9 +305,12 @@ def keypress(self, size: Tuple[int, int], key: str) -> str:
             self.log.clear()
             self.log.extend(self.streams_btn_list)
             self.set_focus('body')
+            self.log.set_focus(self.previous_stream_focus)
             self.view.controller.update_screen()
             return key
-        return super(StreamsView, self).keypress(size, key)
+        return_value = super(StreamsView, self).keypress(size, key)
+        self.previous_stream_focus = self.log.get_focus()[1]
+        return return_value
```

Comment:
The reason i require an extra variable is since the current focus obtained only after the call to
super(StreamsView,self).keypress(size,key)
Hence i need an extra variable to save the return value, store the focus and then return the value.
If possible please suggest how i would have to go about reordering this !

Reply:
That sounds like a good reason to not be able to do that :)